JetSubStructureUtils/Root/BoostedXbbTag.cxx

```diff
@@ -391,22 +396,28 @@ int BoostedXbbTag::result(const xAOD::Jet& jet, std::string algorithm_name, cons
     if(problemWithParent && m_debug) printf("Parent decoration does not exist. ");
     if(!parentEL.isValid() && m_debug) printf("Parent link is not valid. ");
     if(m_debug) printf("\r\n");
-    return -2; // do not fallback
-    /*
-    if(m_debug) printf("Get track jets from groomed jet.\r\n");
-    if(!jet.getAssociatedObjects<xAOD::Jet>("GhostAntiKt2TrackJet", associated_trackJets)){
-      if(m_verbose) printf("<%s>: No associated track jets found on jet.\r\n", APP_NAME);
-      return -2;
-    }
-    if(m_verbose) printf("<%s>: Got track jets from groomed jet.\r\n", APP_NAME);
-    */
+    return -2; // do not fallback -- comment out to enable fallback
+    //// fallback here
+    //if(m_debug) printf("Get track jets from groomed jet.\r\n");
+    //if(!jet.getAssociatedObjects<xAOD::Jet>("GhostAntiKt2TrackJet", associated_trackJets)){
+    //  if(m_verbose) printf("<%s>: No associated track jets found on jet.\r\n", APP_NAME);
+    //  return -2;
+    //}
+    //else if(m_verbose) printf("<%s>: Got track jets from groomed jet.\r\n", APP_NAME);
   } else {
     const xAOD::Jet* parentJet = *parentEL;
     if(!parentJet->getAssociatedObjects<xAOD::Jet>("GhostAntiKt2TrackJet", associated_trackJets)){
       if(m_verbose) printf("<%s>: No associated track jets found on parent jet.\r\n", APP_NAME);
-      return -2;
+      return -2; // do not fallback -- comment out to enable fallback
+      //// fallback here
+      //if(m_debug) printf("Get track jets from groomed jet.\r\n");
+      //if(!jet.getAssociatedObjects<xAOD::Jet>("GhostAntiKt2TrackJet", associated_trackJets)){
+      //  if(m_verbose) printf("<%s>: No associated track jets found on jet.\r\n", APP_NAME);
+      //  return -2;
+      //}
+      //else if(m_verbose) printf("<%s>: Got track jets from groomed jet.\r\n", APP_NAME);
     }
-    if(m_verbose) printf("<%s>: Got track jets from parent jet.\r\n", APP_NAME);
+    else if(m_verbose) printf("<%s>: Got track jets from parent jet.\r\n", APP_NAME);
```

Comment:
Hi Giordon,

The full if statement is

    if(!parentJet->getAssociatedObjects<xAOD::Jet>("GhostAntiKt2TrackJet", associated_trackJets)){
      if(m_verbose) printf("<%s>: No associated track jets found on parent jet.\r\n", APP_NAME);
      return -2; // do not fallback -- comment out to enable fallback
      //// fallback here
      //if(m_debug) printf("Get track jets from groomed jet.\r\n");
      //if(!jet.getAssociatedObjects<xAOD::Jet>("GhostAntiKt2TrackJet", associated_trackJets)){
      //  if(m_verbose) printf("<%s>: No associated track jets found on jet.\r\n", APP_NAME);
      //  return -2;
      //}
      //else if(m_verbose) printf("<%s>: Got track jets from groomed jet.\r\n", APP_NAME);
    }
    else if(m_verbose) printf("<%s>: Got track jets from parent jet.\r\n", APP_NAME);


If fallback is commented out, having an else is optional is because of the return statement; however in case of a problem with ghost-association to the parent, fallback can be enabled by commenting out return and enabling the code under it.  In that case if fallback occurs, we don't want "Got track jets from parent jet" to be printed.  In every case I find that writing "else" conveys the intention better

Concerning this "user comment in/out" instance, and also your three other comments, I agree with you that a better solution would be to implement configuration flags.  In the rush for ICHEP there was no time to be elegant, but please feel free to modify as you see fit.  Another, quick solution would be for me to remove all these comment/options to only keep the default behaviour.

The new recommendations files are included now.

Thanks,
Simon


> Le 22 juil. 2016 à 00:11, Giordon Stark <notifications@github.com> a écrit :
> 
> In JetSubStructureUtils/Root/BoostedXbbTag.cxx:
> 
> >      }
> > -    if(m_verbose) printf("<%s>: Got track jets from parent jet.\r\n", APP_NAME);
> > +    else if(m_verbose) printf("<%s>: Got track jets from parent jet.\r\n", APP_NAME);
> 
> There should be no else here.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>